jolt-core/src/poly/rlc_polynomial.rs

```diff
@@ -324,120 +376,467 @@ impl<F: JoltField> RLCPolynomial<F> {
+    fn build_folded_onehot_tables(&self, eq_k: &[F]) -> VmvFoldedOneHotTables<F> {
+        let k_chunk = eq_k.len();
+        let num_polys = self.instruction_d + self.bytecode_d + self.ram_d;
+        let mut tables: Vec<F> = unsafe_allocate_zero_vec(num_polys * k_chunk);
+
+        for i in 0..self.instruction_d {
+            let coeff = self.instruction_coeffs[i];
+            if coeff.is_zero() {
+                continue;
+            }
+            let base = i * k_chunk;
+            for k in 0..k_chunk {
+                tables[base + k] = coeff * eq_k[k];
+            }
+        }
+
+        let bytecode_base_poly = self.instruction_d;
+        for i in 0..self.bytecode_d {
+            let coeff = self.bytecode_coeffs[i];
+            if coeff.is_zero() {
+                continue;
+            }
+            let base = (bytecode_base_poly + i) * k_chunk;
+            for k in 0..k_chunk {
+                tables[base + k] = coeff * eq_k[k];
+            }
+        }
+
+        let ram_base_poly = self.instruction_d + self.bytecode_d;
+        for i in 0..self.ram_d {
+            let coeff = self.ram_coeffs[i];
+            if coeff.is_zero() {
+                continue;
+            }
+            let base = (ram_base_poly + i) * k_chunk;
+            for k in 0..k_chunk {
+                tables[base + k] = coeff * eq_k[k];
+            }
+        }
+
+        VmvFoldedOneHotTables {
+            k_chunk,
+            instruction_d: self.instruction_d,
+            bytecode_d: self.bytecode_d,
+            ram_d: self.ram_d,
+            tables,
+        }
+    }
+
+    fn new(ctx: &'a StreamingRLCContext<F>) -> Self {
+        let one_hot_params = &ctx.one_hot_params;
+        let log_k_chunk = one_hot_params.log_k_chunk;
+        let instruction_d = one_hot_params.instruction_d;
+        let bytecode_d = one_hot_params.bytecode_d;
+        let ram_d = one_hot_params.ram_d;
+
+        let mut rd_inc_coeff = F::zero();
+        let mut ram_inc_coeff = F::zero();
+        let mut instruction_coeffs = [F::zero(); MAX_INSTRUCTION_D];
+        let mut bytecode_coeffs = [F::zero(); MAX_BYTECODE_D];
+        let mut ram_coeffs = [F::zero(); MAX_RAM_D];
+
+        for (poly_id, coeff) in ctx.dense_polys.iter() {
+            match poly_id {
+                CommittedPolynomial::RdInc => rd_inc_coeff += *coeff,
+                CommittedPolynomial::RamInc => ram_inc_coeff += *coeff,
+                _ => unreachable!("one-hot polynomial found in dense_polys"),
+            }
+        }
+
+        for (poly_id, coeff) in ctx.onehot_polys.iter() {
+            match poly_id {
+                CommittedPolynomial::InstructionRa(idx) => instruction_coeffs[*idx] += *coeff,
+                CommittedPolynomial::BytecodeRa(idx) => bytecode_coeffs[*idx] += *coeff,
+                CommittedPolynomial::RamRa(idx) => ram_coeffs[*idx] += *coeff,
+                _ => unreachable!("dense polynomial found in onehot_polys"),
+            }
+        }
```

Comment:
isn't there already only one coefficient per CommittedPolynomial  in `ctx.dense_polys` and `ctx.onehot_polys`? I don't see what this is for